scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
```

Comment:
ok... Source() is a function, with a defined output, which is actually a map, but defined as a less general interface. The problem with saying "my type conversion will crash" is that, unless the API is unstable, or I'm not actually meant to use the output of Source(), I should be able to use the data that the function outputs. If we take the concept you suggest to the extreme, why not just define everything as an interface{} and make any usable output depend on an explicit type check?